src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
+      } else if (format === 'YY') {
+        const fromIndex = fromFormat.indexOf('YYYY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year.slice(-2));
+      }
+    } else {
+      resultArr.push(dateArr[index]);
+    }
```

Comment:
The logic inside the loop does not handle cases where the `fromFormat` and `toFormat` arrays have different date parts (e.g., converting from `MM-DD-YYYY` to `YYYY-DD-MM`). It would be beneficial to add error handling for unsupported or mismatched date formats.

Reply:
Thank you, this will be useful in the future. In this task, the date formats are given.